client/web/compose/src/components/Admin/Module/UniqueValues.vue

```diff
@@ -0,0 +1,220 @@
+                        {{ $t('none') }}
+                      </b-form-select-option>
+                    </template>
+                  </b-form-select>
+                </td>
+                <td>
+                  <b-form-select
+                    v-model="constraint.multiValue"
+                    :options="multiValueOptions"
+                    :disabled="!constraint.isMulti"
+                    size="sm"
+                  />
+                </td>
+                <td class="text-center">
+                  <c-input-confirm
+                    button-class="text-right"
+                    @confirmed="rule.constraints.splice(consIndex, 1)"
+                  />
+                </td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <b-form-group>
+          <b-input-group>
+            <vue-select
+              v-model="rule.currentField"
+              :placeholder="$t('searchFields')"
+              :options="filterFieldOptions(rule)"
+              :get-option-label="getOptionLabel"
+              class="bg-white"
+            />
+
+            <b-input-group-append>
+              <b-button
+                variant="primary"
+                class="px-4"
+                @click="updateRuleConstraint(rule)"
+              >
+                {{ $t("add") }}
+              </b-button>
+            </b-input-group-append>
+          </b-input-group>
+        </b-form-group>
+      </div>
+    </div>
+
+    <hr>
+
+    <div class="d-flex justify-content-end">
+      <b-button
+        size="lg"
+        variant="link"
+        class="d-flex align-items-center text-decoration-none p-0 mt-3"
+        @click="addNewConstraint"
+      >
+        <font-awesome-icon
+          :icon="['fas', 'plus']"
+          class="mr-2"
+        />
+        {{ $t("addNewConstraint") }}
+      </b-button>
+    </div>
+  </div>
+</template>
+
+<script>
+import { compose } from '@cortezaproject/corteza-js'
+import { VueSelect } from 'vue-select'
+
+export default {
+  i18nOptions: {
+    namespaces: 'module',
+    keyPrefix: 'edit.config.uniqueValues',
+  },
+
+  components: {
+    VueSelect,
+  },
+
+  props: {
+    module: {
+      type: compose.Module,
+      required: true,
+    },
+  },
+
+  computed: {
+    rules: {
+      get () {
+        return this.module.config.recordDeDup.rules
+      },
+      set (value) {
+        this.module.config.recordDeDup.rules = value
+      },
+    },
+
+    modifierOptions () {
+      return [
+        { value: 'ignore-case', text: this.$t('ignoreCase') },
+        { value: 'fuzzy-match', text: this.$t('fuzzyMatch') },
+        { value: 'sounds-link', text: this.$t('soundsLike') },
+      ]
+    },
+
+    multiValueOptions () {
+      return [
+        { value: 'one-of', text: this.$t('oneOf') },
+        { value: 'equal', text: this.$t('equal') },
+      ]
+    },
+  },
+
+  methods: {
+    addNewConstraint () {
+      this.rules.push({
+        name: 'case-sensitive',
```

Comment:
Vivek confirmed the name property is not needed anymore so you can remove it from here